version: '3.3'
services:
  mariadb:
    image: 'docker.io/bitnami/mariadb:10.3-debian-10'
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
      - MARIADB_SQL_MODE=STRICT_ALL_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
      - MARIADB_EXTRA_FLAGS=--local-infile=0 --max-allowed-packet=32M --innodb-buffer-pool-size=256MB
    volumes:
      - 'mariadb_data:/bitnami/mariadb'
    networks:
      - phabri-backend
  phabricator:
    image: 'docker.io/bitnami/phabricator:2021-debian-10'
    networks:
      - traefik-net
      - phabri-backend
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
      - PHABRICATOR_HOST=phabricator.traefik.oninhome.io
    volumes:
      - 'phabricator_data:/bitnami/phabricator'
    deploy:
      labels:
        - "traefik.enable=true"
        - "traefik.http.router.phabricator.rule=Host(`phabricator.traefik.oninhome.io`)"
        - "traefik.http.router.phabricator.entrypoints=web"
        - "traefik.http.services.phabricator.loadbalancer.server.port=8080"
      placement:
        constraints: [node.role==worker]
    depends_on:
      - mariadb
volumes:
  mariadb_data:
    external: false 
  phabricator_data:
    external: false 

networks:
  phabri-backend:
    external: true
